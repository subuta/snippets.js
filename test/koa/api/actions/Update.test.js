/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import UpdateAction from 'lib/koa/api/actions/Update'

describe('knex/api/actions/Update', () => {
  it('should Update koa action', () => {
    expect(format(UpdateAction({
      model: 'user',
      config: {}
    }))).toMatchSnapshot()
  })
})
