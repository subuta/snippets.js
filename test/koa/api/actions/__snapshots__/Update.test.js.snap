// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`knex/api/actions/Update should Update koa action 1`] = `
"user.put('/:id', async (ctx) => {
  const {User} = ctx.state.models
  const {user} = ctx.request.body
  const {sub} = ctx.state.user

  // update specified user.
  const params = {}

  /* mat Before update [start] */
  /* mat Before update [end] */

  ctx.body = await User.query()
    .patchAndFetchById(ctx.params.id, {
      ...user,
      ...params
    })
    .eager('')
})
"
`;
